pkg/config/config.go

```diff
@@ -48,3 +51,48 @@ func NewKVConfiguration(fileName string) (KVConfiguration, error) {
 	err = decoder.Decode(&configuration)
 	return configuration, err
 }
+
+func (c *KVConfiguration) GetReplications() []string {
+	var localStores []string
+	var neighborStores []string
+	var remoteStores []string
+	for _, store := range c.Stores {
+		switch region := store.RegionType; region {
+		case "local":
+			localStores = append(localStores, fmt.Sprintf("%s:%d", store.Host, store.Port))
+		case "neighbor":
+			neighborStores = append(neighborStores, fmt.Sprintf("%s:%d", store.Host, store.Port))
+		case "remote":
+			remoteStores = append(remoteStores, fmt.Sprintf("%s:%d", store.Host, store.Port))
+		}
+	}
+	n := len(localStores)
+	replications := make([]string, n)
+	for idx := 0; idx < n; idx++ {
+		total := c.ReplicaNum
+		replics := make([]string, total)
+		total--
+		if remote := selectRandom(remoteStores); remote != "" {
+			replics[total] = remote
+			total--
+		}
+		if neighbor := selectRandom(neighborStores); neighbor != "" {
+			replics[total] = neighbor
+			total--
+		}
+		for i := 0; i < total+1; i++ {
+			replics[total] = localStores[(idx+i)%n]
+			total--
+		}
+		replications[idx] = strings.Join(replics, ",")
```

Comment:
consider using proper type/objects instead of string everywhere

Reply:
Those store information is from the config files which are all string objects. We have to use string functions to process as designed.